File: Caesar/caesar.py
# get command line arguments from system library
from sys import argv
# get method get_string from module cs50
from cs50 import get_string


def main():
    # get key from command line input
    key = get_argv()
    # get text from user
    plain_text = get_string("plaintext: ")

    print_cipher(plain_text, key)


def print_cipher(text, key):
    print("ciphertext: ", end="")
    for c in text:
        if c.isalpha():
            # get ascii value of char
            temp = ord(c)
            temp += key

            # cipher uppercase char
            if c.isupper():
                while True:
                    if temp >= 65 and temp <= 90:
                        break
                    else:
                        temp = (temp % 90) + 64
                print(chr(temp), end="")

            # cipher lowercase char
            if c.islower():
                while True:
                    if temp >= 97 and temp <= 122:
                        break
                    else:
                        temp = (temp % 122) + 96

                print(chr(temp), end="")
        else:
            print(c, end="")

    print()


def get_argv():
    if len(argv) == 2:
        i = int(argv[1])
        if i >= 0:
            return i
    print("Usage: python caesar.py key")
    exit(1)


if __name__ == "__main__":
    main()

File: caesar/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <stdlib.h>

// global variables
int KEY = 0;
int UPPER = 65;
int LOWER = 97;
int NUM_ALPHA = 26;

// prototypes for functions
bool check_int(string input_key);
void cipher(string input);


int main(int argc, string argv[])
{
    string input_key = argv[1];

    // check command line
    if (argc == 2 && check_int(input_key))
    {
        string input_text = get_string("plaintext: ");
        cipher(input_text);
    }
    else
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
}


// encrypt and print cipher text
void cipher(string input)
{
    // printf("---- in cipher() ----\n");

    string plain_text = input;
    string cipher_text = input;
    int len = strlen(input);

    printf("ciphertext: ");

    for (int i = 0; i < len; i++)
    {
        if (isalpha(plain_text[i]))
        {
            if (isupper(plain_text[i]))
            {
                cipher_text[i] = ((plain_text[i] + KEY - UPPER) % NUM_ALPHA) + UPPER;
            }
            else
            {
                cipher_text[i] = ((plain_text[i] + KEY - LOWER) % NUM_ALPHA) + LOWER;
            }
        }

        printf("%c", cipher_text[i]);
    }

    printf("\n");
}


bool check_int(string input_key)
{
    // printf("---- in check_int() ----\n");

    int len = strlen(input_key);
    string key = input_key;

    for (int i = 0; i < len; i++)
    {
        if (!isdigit(key[i]))
        {
            return false;
        }
    }

    KEY = atoi(key);
    // printf("KEY: %i\n", KEY);

    return true;
}